lib/segment/src/spaces/metric_f16/arm.c

```diff
@@ -0,0 +1,135 @@
+#if !defined PC_VER
+#include <arm_neon.h>
+#else
+#include "NEONvsSSE.h"
+#endif
+
+
+float32_t dotProduct_half_4x4(const float16_t* pSrcA, const float16_t* pSrcB, uint32_t blockSize)
+{
+    float32_t dotProduct = 0.0f;
+    float16x8_t sum1 = vdupq_n_f16(0.0f);
+    float16x8_t sum2 = vdupq_n_f16(0.0f);
+    float16x8_t sum3 = vdupq_n_f16(0.0f);
+    float16x8_t sum4 = vdupq_n_f16(0.0f);
+    uint32_t i = 0;
```

Comment:
Why do you want to define iterator here instead of `for (int i, ...)`?

Reply:
Added the iterator into the for loop